framework/include/mediaplayer/mediaplayer.h

```diff
@@ -0,0 +1,215 @@
+#ifndef __MEDIAPLAYER_H__
+#define __MEDIAPLAYER_H__
+
+#include <queue.h>
+#include <sys/types.h>
+
+/***
+ * Errors
+ */
+#define MEDIA_OK					0
+#define MEDIA_ERROR					1
+#define MEDIA_ERROR_THREAD_INIT		2
+#define MEDIA_ERROR_PCM_WRITE		3
+#define MEDIA_ERROR_WRITE_TO_FILE	4
+#define MEDIA_ERROR_CREATE			5
+#define MEDIA_ERROR_UNKNOWN_FILE	6
+
+/***
+ * Enumerations
+ */
+typedef enum {
```

Comment:
I think we don't have to distinguish pcm & wav.
Because wav is pcm data + header data.
First check file's extension, and try to read header.
Then we can decide file type.
Also if it support other media type, will you add 2 types for all media type?
I think we'd better use below
#define MEDIA_TYPE_PCM 0x00000001
#define MEDIA_TYPE_2       0x00000002
#define MEDIA_TYPE_3       0x00000003
.
.#define MEDIA_TYPE_MASK 0x0000ffff

#define MEDIA_PLAY_TYPE_1   0x00010000
#define MEDIA_PLAY_TYPE_STREAM 0x00020000
#define MEDIA_PLAY_TYPE_MASK 0xffff0000

static int get_media_type(int type)  {
   return type & MEDIA_TYPE_MASK;
}
static int get_play_type(int type) {
    return type & MEDIA_PLAY_TYPE_MASK;
}

Reply:
These flags are just to specify whether we got a PCM file or a WAV file. This flag can be useful when we are recording data and give output according to the file type. In PCM type we do not need wav header. 

Yes I will change the flags into two parts, file type and media type.